epsagon/events/pymongo.py

```diff
@@ -33,30 +35,34 @@ def __init__(self, wrapped, instance, args, kwargs, start_time, response,
         :param response: response data
         :param exception: Exception (if happened)
         """
-
         super(PyMongoEvent, self).__init__(start_time)
 
-        documents = args[0]
+        self.resource['operation'] = getattr(wrapped, '__name__')
 
-        self.event_id = 'mongo-{}'.format(str(uuid4()))
-        self.resource['name'] = instance.full_name
+        if not args:
+            documents = list()
 
-        self.resource['operation'] = \
-            'insert_many' if isinstance(documents, list) else 'insert_one'
+        else:
+            documents = args[0]
+
+        self.event_id = 'mongo-{}'.format(str(uuid4()))
+        self.resource['name'] = instance.name
         address = list(getattr(
             instance.database.client,
             '_topology_settings'
-        ).seeds)[0]
-
-        if self.resource['operation'] == 'insert_one':
-            documents = [documents]
+            ).seeds)[0]
 
         self.resource['metadata'] = {
-            'DB URL': ':'.join([str(x) for x in address]),
-            'DB Name': str(instance.database.name),
-            'Collection Name': str(instance.name),
-        }
-        add_data_if_needed(self.resource['metadata'], 'Items', documents)
+                'DB URL': ':'.join([str(x) for x in address]),
+                'DB Name': str(instance.database.name),
+                'Collection Name': str(instance.collection.name),
+            }
+
+        if self.resource['operation'] in PyMongoEvent.INSERT_OPERATIONS:
+            add_data_if_needed(self.resource['metadata'], 'Items', documents)
```

Comment:
I think there's an issue with the document/documents logic (regarding the insert one/many operations)
Also, the documents param is only relevant for specific operations. So, I think I would add some payload colelction here according to the operation (insert many > so it documents, insert one/update one > document, etc)